Update Serialization to Core M1

sdelamo opened the pull request.

sdelamo committed:
remove groovy version

Use version from Micronaut Build

sdelamo committed:
remove kotlin and ksp version

use version from micronaut build

sdelamo committed:
micronaut core 5.0.0-M1

sdelamo committed:
remove micronaut-docs version

sdelamo committed:
remove micronaut-test version

sdelamo committed:
define parent id module

sdelamo committed:
add back launcher

sdelamo committed:
Use JSpecify annotations

dstepanov committed:
Fix bean context

sdelamo committed:
fix jspecify annotation placement

sdelamo committed:
Jackson 2 to Jackson 3

sdelamo commented:
@yawkat please, check this issue

yawkat committed:
progress

yawkat committed:
remove TypeThrashingTest (see core)

sdelamo committed:
micronaut-platform = "4.10.3"

sdelamo committed:
migrate to Jackson 3

sdelamo committed:
Jackson 3

sdelamo committed:
replace tools.jackson.core.JsonParseException with tools.jackson.core.JacksonException

sdelamo committed:
JsonMappingException (root for databind exceptions) becomes DatabindException

sdelamo committed:
fix spock version

sdelamo committed:
kotlin nullability changes

sdelamo committed:
use ksp plugin not kapt

sdelamo committed:
use micronaut nullability not jspecify

sdelamo committed:
objectMapper::setFilter provider does not exist

sdelamo committed:
update DatabindPredicateFilter

sdelamo commented on test-suite-tck-jackson-databind/src/test/groovy/io/micronaut/serde/tck/jackson/databind/DatabindFiltersRegistrar.java:
```diff
@@ -14,7 +14,7 @@ public class DatabindFiltersRegistrar {
         SimpleFilterProvider filterProvider = new SimpleFilterProvider();
         filterProvider.addFilter("ignore-value", predicateFilter);
-        objectMapper.setFilterProvider(filterProvider);
+        //TODO objectMapper.setFilterProvider(filterProvider);
```
@yawkat @dstepanov `ObjectMappper::setFilterProvider` no longer exists.

yawkat committed:
checkstyle

yawkat committed:
propagate JacksonException

yawkat committed:
update JacksonConfigurationSpec

yawkat committed:
remove newJavaParser override

yawkat committed:
annotation migration

yawkat committed:
disable FAIL_ON_NULL_FOR_PRIMITIVES for DatabindBasicSerdeSpec

dstepanov commented on serde-support/src/main/java/io/micronaut/serde/support/deserializers/ErrorCatchingDeserializer.java:
```diff
@@ -55,6 +66,9 @@ public T deserialize(Decoder decoder, DecoderContext context, Argument<? super T
         } catch (IOException e) {
             throw e;
         } catch (Exception e) {
+            if (JACKSON_EXCEPTION_CLASS != null && JACKSON_EXCEPTION_CLASS.isInstance(e)) {
```
Can we have this configurable?

yawkat replied:
I think we should find another solution anyway, I made an issue for it https://github.com/micronaut-projects/micronaut-serialization/issues/1212

sdelamo committed:
use setJsonReadFeatures(Map.of(JsonReadFeature.ALLOW_JAVA_COMMENTS

Replace:

```java
SerdeJacksonConfiguration newConf = new SerdeJacksonConfiguration()
newConf.setParserFeatures(Map.of(JsonParser.Feature.ALLOW_COMMENTS, true))
```

with:

```java
SerdeJacksonConfiguration newConf = new SerdeJacksonConfiguration()
newConf.setJsonReadFeatures(Map.of(JsonReadFeature.ALLOW_JAVA_COMMENTS, true))
```

sdelamo committed:
Use JSONAssert to compare with JSON with non-strict array ordering.

sdelamo committed:
create FormatExceptionClass method

sdelamo committed:
Use JSONAssert to compare JSON with non-strict array ordering.

sdelamo committed:
Use JSONAssert to compare JSON with non-strict array ordering.

sdelamo committed:
Use JSONAssert to compare JSON with non-strict array ordering.

sdelamo committed:
Use JSONAssert to compare JSON with non-strict array ordering.

sdelamo committed:
Use JSONAssert to compare JSON with non-strict array ordering.

sdelamo committed:
Use JSONAssert to compare JSON with non-strict array ordering.

yawkat committed:
support old databind annotations

sdelamo committed:
micronaut = "5.0.0-M2"

sdelamo committed:
Use JSONAssert to compare JSON with non-strict array ordering

sdelamo committed:
fail-on-null-for-primitives=false

sdelamo committed:
set Jackson fail-on-unknown-properties=true

Messages has chagned from

Unrecognized field "$propertyName" (class $className), not marked as ignorable

to

Unrecognized property "$propertyName" (class $className), not marked as ignorable

sdelamo committed:
jackson accept-case-insensitive-enums=true

sdelamo committed:
unknownPropertyMessage or unknownFieldMessage

sdelamo committed:
use new property naming for jackson features

sdelamo committed:
use jackson.mapper-features.

sdelamo committed:
clear properties define it in a test by test basis

sdelamo committed:
fail-on-null-for-primitives false

sdelamo committed:
remove extra curly braces

sdelamo committed:
use stream-read-features.strict-duplicate-detection

sdelamo committed:
skip tests temporarily No value type configured for ObjectReader"

sdelamo committed:
fix assertion

sdelamo committed:
ignore DatabindJsonFilterSpec

sdelamo committed:
ignore DatabindJsonTypeInfoSpec

sdelamo committed:
ignore SerdeJsonSubtypesKotlinSpec
